projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-users-endpoint.php

```diff
@@ -229,12 +229,25 @@ function ( $viewer ) use ( $args ) {
 
 					$combined_users = array_merge( $users, $viewers );
 
-					// When viewers are included, we ignore the order & orderby parameters.
+					// When viewers are included, we try to respect orderby if it's login or email.
 					if ( $include_viewers ) {
+						// First we need to be sure to obtain the viewer emails and logins.
+						foreach ( $combined_users as &$user ) {
+							$wp_user     = new WP_User( $user->ID );
+							$user->email = $wp_user->user_email;
+							$user->login = $wp_user->user_login;
+						}
+
+						$order   = $args['order'] ?? 'asc';
+						$orderby = $args['order_by'] ?? 'name';
+
 						usort(
 							$combined_users,
-							function ( $a, $b ) {
-								return strcmp( strtolower( $a->name ), strtolower( $b->name ) );
+							function ( $a, $b ) use ( $order, $orderby ) {
+								if ( 'desc' === strtolower( $order ) ) {
+									return gmp_neg( strcmp( strtolower( $a->{$orderby} ), strtolower( $b->{$orderby} ) ) );
+								}
+								return strcmp( strtolower( $a->{$orderby} ), strtolower( $b->{$orderby} ) );
```

Comment:
Looks like this is the only way we can do it.

As a follow up, I also think we need to dedupe the combined array for users that are both Viewers and something else. See: https://github.com/Automattic/jetpack/pull/28317#issuecomment-2658129481

Going forward, it might be an idea to remove the Viewer role when a user changes roles. I'm not sure since the Viewer role is special - an Editor might still want to be a Viewer so they can preview the private site, if that's the way it works 🤷🏻

Reply:
Pretty sure any role can preview a private site. I've spent a lot of testing time creating wordpress.com accounts for all my alt emails and adding them to several of my private sites 😂

Reply:
You're right. I think we should immediately revoke the Viewer status when a Viewer's role changes

Mentioned in pdgR5j-1dB-p2